atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/PrismAgent.kt

```diff
@@ -990,6 +1031,128 @@ class PrismAgent {
+            proofs = proofTypes,
+            options = PresentationOptions(jwtAlg = arrayOf("EdDSA"))
+        )
+
+        // TODO: Use type and format to differentiate between presentation request and presentation exchange
+        val attachmentDescriptor = AttachmentDescriptor(
+            mediaType = "application/json",
+            format = CredentialType.JWT.type,
+            data = AttachmentBase64(Json.encodeToString(presentationDefinitionRequest).base64UrlEncoded)
+        )
+
+        val presentationRequest = EdgePresentationRequest(
+            body = EdgePresentationRequest.Body(),
+            attachments = arrayOf(attachmentDescriptor),
+            thid = UUID.randomUUID4().toString(),
+            from = newPeerDID,
+            to = toDID
+        )
+
+        connectionManager.sendMessage(presentationRequest.makeMessage())
+    }
+
+    suspend fun handlePresentationDefinitionRequest(msg: Message, credential: Credential) {
+        if (credential::class != JWTCredential::class) {
+            throw PrismAgentError.InvalidCredentialError(credential)
+        }
+
+        val edgePresentationRequest = EdgePresentationRequest.fromMessage(msg)
+        val msgAttachmentDescriptor =
+            edgePresentationRequest.attachments.find { it.data::class == AttachmentBase64::class }
+                ?: throw PrismAgentError.AttachmentTypeNotSupported()
+
+        val attachmentBase64 = msgAttachmentDescriptor.data as AttachmentBase64
+        val presentationDefinitionRequest =
+            Json.decodeFromString<PresentationDefinition>(attachmentBase64.base64.base64UrlDecoded)
+
+        val didString =
+            credential.subject ?: throw Exception("Credential must contain subject")
+
+        val privateKeyKeys = pluto.getDIDPrivateKeysByDID(DID(didString)).first()
+        val privateKey = privateKeyKeys.first()
+            ?: throw PrismAgentError.CannotFindDIDPrivateKey(didString)
+
+        val presentationSubmissionProof = pollux.createPresentationSubmission(
+            presentationDefinitionRequest = presentationDefinitionRequest,
+            credential = credential,
+            did = DID(didString),
+            privateKey = privateKey
+        )
+
+        val attachmentDescriptor = AttachmentDescriptor(
+            mediaType = "application/json",
+            format = CredentialType.JWT.type,
+            data = AttachmentBase64(Json.encodeToString(presentationSubmissionProof).base64UrlEncoded)
+        )
+        val presentationSubmission = PresentationSubmission(
+            body = PresentationSubmission.Body(),
+            attachments = arrayOf(attachmentDescriptor),
+            thid = edgePresentationRequest.thid,
+            from = edgePresentationRequest.to,
+            to = edgePresentationRequest.from
+        )
+        connectionManager.sendMessage(presentationSubmission.makeMessage())
+    }
+
+    suspend fun handlePresentationSubmission(msg: Message, withResolver: DIDResolver): Boolean {
```

Comment:
Same in here you don't need the resolver since PrismAgent already got a Castor internally

Reply:
Good point